packages/pluggableWidgets/slider-web/src/Slider.tsx

```diff
@@ -0,0 +1,38 @@
+import { createElement } from "react";
+import { SliderContainerProps } from "../typings/SliderProps";
+import { getMinValue } from "./utils/getMinValue";
+import { getMaxValue } from "./utils/getMaxValue";
+import { getStepValue } from "./utils/getStepValue";
+import { useMarks } from "./utils/useMarks";
+import { createHandleGenerator } from "./utils/createHandleGenerator";
+import { useOnChangeDebounced } from "./utils/useOnChangeDebounced";
+import { Slider as SliderComponent } from "./components/Slider";
+import "rc-slider/assets/index.css";
+import "rc-tooltip/assets/bootstrap.css";
+import "./ui/Slider.scss";
+import { isVertical } from "./utils/isVertical";
+
+export default function Slider(props: SliderContainerProps) {
+    const { valueAttribute, class: className, style } = props;
+    const minValue = getMinValue(props);
+    const maxValue = getMaxValue(props);
+    const stepValue = getStepValue(props);
+    const marks = useMarks(props);
+    const handle = createHandleGenerator(props);
+    const { onChange } = useOnChangeDebounced(props);
```

Comment:
I don't like the fact that we're passing the entire `props` object to every function. Every function only needs like 2/3 props so it feels unnecessary. Now it's also impossible to understand what the function needs and will do without actually going into the function because they could be using any prop. Being more specific means we also have to be more specific here and will read more intuitively. So for example: 
`getMinValue({ minValueType: props.minValueType, staticMinimumValue: props.staticMinimumValue, minAttribute: props.minAttribute, expressionMinimumValue: props.expressionMinimumValue });` 
or even 
`getMinValue({ minValueType, staticMinimumValue, minAttribute, expressionMinimumValue });` if you destructure them

Reply:
I do see the purpose of passing specific properties, but based on the examples you gave above seems less readable. As props is just a reference for the object there is no performance issue passing it through the functions.

Reply:
It might be less readable in the sense that it's longer, but it becomes more readable in the sense that you know what the function needs when reading this `src/Slider.tsx` file. Creating functions that accept the entire `props` as parameter means that you can rarely reuse them. For this situation is fine because we only use them at this location, but I don't think it's a great pattern.

Reply:
🙀